src/RMP/Translate/TranslateFactory.php

```diff
@@ -137,9 +137,10 @@ protected function getTranslator($locale, $messageSelector, $cachepath, $debug)
     protected function getFilePath($basePath, $locale, $domain)
     {
         // Strip the _GB etc. bit off the locale for portability
-        if (strlen($locale) > 3) {
-            $locale = substr($locale, 0, -strlen(strrchr($locale, '_')));
+        if (strpos($locale, '_') !== false) {
```

Comment:
Is it worth moving this code in to a well named method? It's not immediately clear what it's doing.
Also, I think it's sufficiently complex to need a test along with it.